lib/project_types/script/layers/infrastructure/languages/task_runner.rb

```diff
@@ -5,14 +5,30 @@ module Layers
     module Infrastructure
       module Languages
         class TaskRunner
-          TASK_RUNNERS = {
-            "assemblyscript" => AssemblyScriptTaskRunner,
-            "typescript" => TypeScriptTaskRunner,
-          }
-
           def self.for(ctx, language, script_name)
-            raise Errors::TaskRunnerNotFoundError unless TASK_RUNNERS[language]
-            TASK_RUNNERS[language].new(ctx, script_name)
+            task_runner = {
+              "assemblyscript" => AssemblyScriptTaskRunner,
+              "typescript" => TypeScriptTaskRunner,
+            }
+            raise Errors::TaskRunnerNotFoundError unless task_runner[language]
+            task_runner[language].new(ctx, script_name)
+          end
+
+          def check_system_dependencies!
+            raise NotImplementedError
+          end
+
+          def check_tool_version!(tool, min_required_version)
```

Comment:
This should be a `protected` method, but I see you changed that in https://github.com/Shopify/shopify-cli/pull/1634